src/sys/socket/mod.rs

```diff
@@ -430,11 +448,12 @@ impl<'a> ControlMessage<'a> {
                 copy_bytes(fds, buf);
             },
             ControlMessage::ScmTimestamp(t) => {
-                let cmsg = cmsghdr {
-                    cmsg_len: self.len() as type_of_cmsg_len,
-                    cmsg_level: libc::SOL_SOCKET,
-                    cmsg_type: libc::SCM_TIMESTAMP,
-                    cmsg_data: [],
+                let cmsg = {
+                  let mut cmsg: cmsghdr = mem::uninitialized();
```

Comment:
Shouldn't think be `mem::zero()` so we make sure all memory is in a known state?